### PR TITLE
handled cloning weblibs failure

### PR DESCRIPTION
**Pull request by dmahmouali**
issue: https://github.com/threefoldtech/js-sdk/issues/103

### CODE REVIEW

File: jumpscale/packages/weblibs/package.py [hunk 1]
```diff
@@ -11,7 +11,15 @@ def install(self):
         """Called when package is added
         """
         if not j.sals.fs.exists(j.sals.fs.join_paths(self.path, "jumpscaleX_weblibs")):
-            j.tools.git.clone_repo(url=self.url, dest=self.path, branch_or_tag=self.branch)
+            retries = 5
```
Comment (waleedhammam) [inline jumpscale/packages/weblibs/package.py:14]: 5 is too much let's make it 2 :D

File: jumpscale/packages/weblibs/package.py [hunk 2]
```diff
@@ -11,7 +11,15 @@ def install(self):
...
+            while retries:
+                try:
+                    j.tools.git.clone_repo(url=self.url, dest=self.path, branch_or_tag=self.branch)
```
Comment (waleedhammam) [inline jumpscale/packages/weblibs/package.py:17]: can make a shallow cloning to reduce cloning size depth=1 for example

File: jumpscale/packages/weblibs/package.py (hunk 1 above)
Comment (waleedhammam) [inline jumpscale/packages/weblibs/package.py:14]: 5 is too much let's make it 3

File: jumpscale/packages/weblibs/package.py (hunk 2 above)
Comment (dmahmouali) [reply jumpscale/packages/weblibs/package.py]: added, thanks

File: jumpscale/packages/weblibs/package.py (hunk 1 above)
Comment (xmonader) [reply jumpscale/packages/weblibs/package.py]: what happens if it failed and the dir got created and not a real git repo?

Comment (waleedhammam) [reply jumpscale/packages/weblibs/package.py]: git doesn't create the repo file unless it's completed. no ?

File: jumpscale/packages/weblibs/package.py [hunk 3]
```diff
@@ -11,7 +11,15 @@ def install(self):
...
         """
         if not j.sals.fs.exists(j.sals.fs.join_paths(self.path, "jumpscaleX_weblibs")):
-            j.tools.git.clone_repo(url=self.url, dest=self.path, branch_or_tag=self.branch)
+            retries = 2
+            while retries:
+                try:
+                    j.tools.git.clone_repo(url=self.url, dest=self.path, branch_or_tag=self.branch, depth=1)
+                    break
+                except Exception:
+                    if retries == 1:
```
Comment (waleedhammam) [inline jumpscale/packages/weblibs/package.py:20]: if retries:

Comment (dmahmouali) [reply jumpscale/packages/weblibs/package.py:20]: but what for while condition ?

File: jumpscale/packages/weblibs/package.py [hunk 4]
```diff
@@ -11,7 +11,15 @@ def install(self):
...
+                        raise j.exceptions.Timeout("Clone weblibs repo failed")
+                    retries = retries - 1
```
Comment (waleedhammam) [inline jumpscale/packages/weblibs/package.py:22]: retries -= 1

File: jumpscale/packages/weblibs/package.py (hunk 3 above)
Comment (waleedhammam) [reply jumpscale/packages/weblibs/package.py:20]: let's make it after while then ?

File: jumpscale/packages/weblibs/package.py (hunk 1 above)
Comment (dmahmouali) [reply jumpscale/packages/weblibs/package.py]: tried cloning with wrong branch and anther with no internet connection and no dirs created :+1:

**Pull request merged**